web-components/toggle/toggle.svelte

```diff
@@ -0,0 +1,212 @@
+  let dragStartX: number | undefined
+  let dragOffsetX: number = 0
+
+  const dispatch = createEventDispatcher()
+  const toggle = (newValue?: boolean) => {
+    if (newValue === undefined) newValue = !on
+    on = newValue
+    dispatch('toggle', { on })
+  }
+</script>
+
+<label class={`leo-toggle size-${size}`}>
+  <button
+    on:mousedown={(e) => {
+      if (disabled) return
+
+      handledClick = true
+      dragStartX = e.clientX
+    }}
+    on:click={(e) => {
+      if (handledClick || disabled) {
+        handledClick = false
+        return
+      }
+
+      toggle()
+    }}
+    {disabled}
+    role="switch"
+    aria-checked={on}
+  >
+    <div
+      bind:this={thumb}
+      class="thumb"
+      class:dragging={!!dragOffsetX}
+      aria-hidden="true"
+      style="--drag-offset: {dragOffsetX}px"
+    >
+      <div class="on-icon">
+        <slot name="on-icon">
+          <svg
+            width="18"
+            height="18"
+            viewBox="0 0 18 18"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path
+              d="M14.6461 4.00015C14.2516 3.69959 13.6882 3.77573 13.3876 4.17021L7.47971 11.9243L4.53279 8.9774C4.18211 8.62672 3.61356 8.62672 3.26288 8.9774C2.91221 9.32807 2.91221 9.89663 3.26288 10.2473L6.93634 13.9208C7.11956 14.104 7.37301 14.1991 7.63155 14.1817C7.89009 14.1643 8.12851 14.0361 8.28555 13.83L14.8161 5.25861C15.1167 4.86413 15.0406 4.3007 14.6461 4.00015Z"
+              fill="currentColor"
+            />
+          </svg>
+        </slot>
+      </div>
+    </div>
+  </button>
+  <slot></slot>
+</label>
+<svelte:window
+  on:mouseup={() => {
+    // If we didn't receive a mouse down, there's nothing to do.
+    if (dragStartX === undefined) return
+
+    // If we didn't drag just toggle the state.
+    if (dragOffsetX === 0) {
+      toggle()
+    } else {
+      if (dragOffsetX > DRAG_AMOUNT_TO_CHANGE && !on) toggle(true)
+      if (dragOffsetX < -DRAG_AMOUNT_TO_CHANGE && on) toggle(false)
+    }
+
+    // Reset the dragging attributes.
+    dragStartX = undefined
+    dragOffsetX = 0
+  }}
+  on:mousemove={(e) => {
+    if (dragStartX === undefined) return
+    dragOffsetX = e.clientX - dragStartX
+  }}
+/>
+
+<style lang="scss">
+  .leo-toggle {
+    --width: var(--leo-toggle-width, 56px);
+    --height: var(--leo-toggle-height, 32px);
+    --padding: var(--leo-toggle-padding, 2px);
+    --on-color: var(
+      --leo-toggle-on-color,
+      var(--color-interaction-button-primary-background)
+    );
+    --on-color-hover: var(--leo-toggle-on-color-hover, var(--color-primary-60));
+    --off-color: var(--leo-toggle-off-color, var(--color-gray-30));
+    --off-color-hover: var(--leo-toggle-off-color-hover, var(--color-gray-40));
+    --thumb-color: var(--leo-toggle-thumb-color, var(--color-white));
+    --label-gap: var(--leo-toggle-label-gap, var(--spacing-4));
+    --label-flex-direction: var(--leo-toggle-label-flex-direction, row);
```

Comment:
So yeah, we can't put them in a `:host` section because we might not always be in a web-component (and if we aren't our variables won't be defined).

I defined the variables like this so that:
1) The variables have a unique namespace (`--leo-toggle-`)
2) Consumers can override any part of it (don't know if they'll need to)
3) All the colors have a sensible default (where possible, a leo token)